### PR TITLE
Add Core Data persistent history tracking support

### PR DESCRIPTION
**Pull request by samsymons**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1200194497630846/1204326775730087/f
iOS PR: Please drag and drop this into the iOS repo to test - no changes are needed, a PR will be opened once this is merged
macOS PR: Please drag and drop this into the macOS repo to test - no changes are needed, a PR will be opened once this is merged
What kind of version bump will this require?: Minor

**Optional**:

Tech Design URL:
CC: @bwaresiak 

**Description**:

We need to support the option to turn on persistent history tracking on Core Data databases as a part of upcoming feature work.

This option has been added to `CoreDataDatabase` as a false-by-default boolean in the init function.

**Steps to test this PR**:
1. Check that Core Data functionality works as expected (right now it's just bookmarks, so test that adding/removing/etc bookmarks all works)

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Review by amddg44 — APPROVED** [review]
LGTM! I've added, deleted and moved bookmarks, edited their titles and urls, created new folders, re-ordered, added to favorites, removed from favorites and everything worked correctly 👍

**Review by bwaresiak — CHANGES_REQUESTED** [review]

File: Sources/Persistence/CoreDataDatabase.swift [hunk 1]
```diff
@@ -77,14 +77,20 @@ public class CoreDataDatabase: ManagedObjectContextFactory {
...
         self.container = NSPersistentContainer(name: name, managedObjectModel: model)
         self.containerLocation = containerLocation
         
         let description = NSPersistentStoreDescription(url: containerLocation.appendingPathComponent("\(name).sqlite"))
         description.type = NSSQLiteStoreType
         description.isReadOnly = readOnly
+
+        if enablePersistentHistoryTracking {
+            description.setOption(true as NSNumber, forKey: NSPersistentHistoryTrackingKey)
+            description.setOption(true as NSNumber, forKey: NSPersistentStoreRemoteChangeNotificationPostOptionKey)
```
[the remark this answers is not in the document]
Comment (samsymons) [reply Sources/Persistence/CoreDataDatabase.swift:92]: Good point, I'll update this to use parameters

[the remark this answers is not in the document]
Comment (samsymons) [reply Sources/Persistence/CoreDataDatabase.swift]: @bwaresiak Done now, please take another look!

**Review by bwaresiak — APPROVED** [review]
LGTM

**Pull request merged**